components/brave_wallet/browser/simple_hash_client.cc

```diff
@@ -637,4 +1240,76 @@ GURL SimpleHashClient::GetSimpleHashNftsByWalletUrl(
   return url;
 }
 
+// static
+// Creates a URL like
+// https://simplehash.wallet.brave.com/api/v0/nfts/{chain}/{contract_address}/{token_id}
+GURL SimpleHashClient::GetNftUrl(const std::string& token_id,
+                                 const std::string& contract_address,
+                                 const std::string& chain_id,
+                                 mojom::CoinType coin) {
+  // Create the URL
+  const std::optional<std::string> simple_hash_chain_id =
+      ChainIdToSimpleHashChainId(chain_id);
+  if (!simple_hash_chain_id) {
+    return GURL();
+  }
+
+  GURL url;
+  if (coin == mojom::CoinType::SOL) {
+    // We don't include the token ID for solana. Instead we use the contract
+    // address (mint address) only.
+    url = GURL(base::StrCat({kSimpleHashBraveProxyUrl, "/api/v0/nfts/",
+                             *simple_hash_chain_id, "/", contract_address}));
+  } else {
+    url = GURL(base::StrCat({kSimpleHashBraveProxyUrl, "/api/v0/nfts/",
+                             *simple_hash_chain_id, "/", contract_address, "/",
+                             token_id}));
+  }
+
+  return url;
+}
+
+// static
+// Creates a URL like
+// https://simplehash.wallet.brave.com/api/v0/nfts/assets?solana.2iZBbRGnLVEEZH6JDsaNsTo66s2uxx7DTchVWKU8oisR,solana.3knghmwnuaMxkiuqXrqzjL7gLDuRw6DkkZcW7F4mvkK8,ethereum.0xed5af388653567af2f388e6224dc7c4b3241c544.2767
+GURL SimpleHashClient::GetNftsUrl(
+    mojom::CoinType coin,
+    const std::vector<std::string>& chain_ids,
+    const std::vector<std::string>& contract_addresses,
+    const std::vector<std::string>& token_ids) {
+  if (chain_ids.empty() || contract_addresses.empty() || token_ids.empty()) {
+    return GURL();
+  }
+
+  if (chain_ids.size() != contract_addresses.size() ||
+      chain_ids.size() != token_ids.size()) {
+    return GURL();
+  }
```

Comment:
Let's have a mojom struct which encapsulates this tuple of chain_id, contract_address, token_id